components/datadog/apps/common/common.go

```diff
@@ -0,0 +1,23 @@
+package commonapp
+
+import (
+	"github.com/DataDog/test-infra-definitions/common/config"
+	"github.com/pulumi/pulumi-kubernetes/sdk/v4/go/kubernetes"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+type K8SDeployer interface {
+	DeployToKubeProvider(e config.CommonEnvironment, kubeProvider *kubernetes.Provider) (pulumi.ComponentResource, error)
```

Comment:
I feel like the `namespace` and `pulumiOptions` could be in the signature of this function instead of being attributes of the `App` struct. Because we could have one single App that we want to deploy to different `namespace`.
This comment can be ignored, but I just found it more natural to select the namespace and the pulumiOptions when we deploy the App rather than when we create it

Reply:
I initially had `namespace` in the `DeployToKubeProvider` and then moved it to the app instance. I agree the namespace is part of the deployment rather than the app definition, will change code on datadog-agent side to have an `WithAppWorkload` that takes an app and a namespace. It is harder to pass `pulumiOptions` as they depend on resources defined in the run function, I think in that case it is better to define a custom run function